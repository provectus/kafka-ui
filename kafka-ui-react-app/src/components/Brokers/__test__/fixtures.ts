import { BrokersLogdirs } from 'generated-sources';

export const brokersPayload = [
  { id: 1, host: 'b-1.test.kafka.amazonaws.com', port: 9092 },
  { id: 2, host: 'b-2.test.kafka.amazonaws.com', port: 9092 },
];

export const clusterStatsPayload = {
  brokerCount: 2,
  activeControllers: 1,
  onlinePartitionCount: 138,
  offlinePartitionCount: 0,
  inSyncReplicasCount: 239,
  outOfSyncReplicasCount: 0,
  underReplicatedPartitionCount: 0,
  diskUsage: [
    { brokerId: 0, segmentSize: 334567, segmentCount: 245 },
    { brokerId: 1, segmentSize: 12345678, segmentCount: 121 },
  ],
  version: '2.2.1',
};

const partition = {
  broker: 2,
  offsetLag: 0,
  partition: 2,
  size: 0,
};
const topics = {
  name: '_confluent-ksql-devquery_CTAS_NUMBER_OF_TESTS_59-Aggregate-Aggregate-Materialize-changelog',
  partitions: [partition],
};

export const brokerLogDirsPayload: BrokersLogdirs[] = [
  {
    error: 'NONE',
    name: '/opt/kafka/data-0/logs',
    topics: [
      {
        ...topics,
        partitions: [partition, partition, partition],
      },
      topics,
      {
        ...topics,
        partitions: [],
      },
    ],
  },
];
