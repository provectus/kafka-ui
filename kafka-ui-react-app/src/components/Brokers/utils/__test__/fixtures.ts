import { BrokerMetrics } from 'generated-sources';

export const brokerMetricsPayload: BrokerMetrics = {
  segmentSize: 23,
  segmentCount: 23,
  metrics: [
    {
      name: 'TotalFetchRequestsPerSec',
      labels: {
        canonicalName:
          'kafka.server:name=TotalFetchRequestsPerSec,topic=_connect_status,type=BrokerTopicMetrics',
      },
      value: 10,
    },
    {
      name: 'ZooKeeperRequestLatencyMs',
      value: 11,
    },
    {
      name: 'RequestHandlerAvgIdlePercent',
    },
  ],
};
export const transformedBrokerMetricsPayload =
  '{"segmentSize":23,"segmentCount":23,"metrics":[{"name":"TotalFetchRequestsPerSec","labels":{"canonicalName":"kafka.server:name=TotalFetchRequestsPerSec,topic=_connect_status,type=BrokerTopicMetrics"},"value":10},{"name":"ZooKeeperRequestLatencyMs","value":11},{"name":"RequestHandlerAvgIdlePercent"}]}';
