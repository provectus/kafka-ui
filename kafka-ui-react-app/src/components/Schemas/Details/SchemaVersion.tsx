import React from 'react';
import { SchemaSubject } from 'generated-sources';
import JSONEditor from 'components/common/JSONEditor/JSONEditor';

interface SchemaVersionProps {
  version: SchemaSubject;
}

const SchemaVersion: React.FC<SchemaVersionProps> = ({
  version: { version, id, schema },
}) => {
  const [isOpen, setIsOpen] = React.useState(false);
  const toggleIsOpen = () => setIsOpen(!isOpen);
  return (
    <tr>
      <td>
        <span
          className="icon has-text-link is-size-7 is-small is-clickable"
          onClick={toggleIsOpen}
          aria-hidden
        >
          <i className={`fas fa-${isOpen ? 'minus' : 'plus'}`} />
        </span>
      </td>
      <td>{version}</td>
      <td>{id}</td>
      <td
        className="has-text-overflow-ellipsis is-family-code"
        style={{ width: '100%', maxWidth: 0 }}
      >
        {isOpen ? (
          <JSONEditor
            isFixedHeight
            name="schema"
            value={
              schema.trim().startsWith('{')
                ? JSON.stringify(JSON.parse(schema), null, '\t')
                : schema
            }
            setOptions={{
              showLineNumbers: false,
              maxLines: 40,
            }}
            readOnly
          />
        ) : (
          <span>{schema}</span>
        )}
      </td>
    </tr>
  );
};

export default SchemaVersion;
