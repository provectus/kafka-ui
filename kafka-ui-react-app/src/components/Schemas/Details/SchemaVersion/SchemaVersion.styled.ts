import styled from 'styled-components';

export const Wrapper = styled.tr`
  background-color: ${({ theme }) => theme.schema.backgroundColor.tr};
  & > td {
    padding: 16px !important;
    & > div {
      background-color: ${({ theme }) => theme.schema.backgroundColor.div};
      border-radius: 8px;
      padding: 24px;
    }
  }
`;
