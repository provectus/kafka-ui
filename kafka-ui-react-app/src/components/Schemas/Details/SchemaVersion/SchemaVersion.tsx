import React from 'react';
import EditorViewer from 'components/common/EditorViewer/EditorViewer';
import { SchemaSubject } from 'generated-sources';
import { Row } from '@tanstack/react-table';

interface Props {
  row: Row<SchemaSubject>;
}

const SchemaVersion: React.FC<Props> = ({ row }) => {
  return (
    <EditorViewer
      data={row?.original?.schema}
      schemaType={row?.original?.schemaType}
    />
  );
};

export default SchemaVersion;
