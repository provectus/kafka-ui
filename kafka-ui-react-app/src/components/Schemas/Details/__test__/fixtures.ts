import { SchemaSubject, SchemaType } from 'generated-sources';
import {
  schemaVersion1,
  schemaVersion2,
  schemaVersionWithNonAsciiChars,
} from 'redux/reducers/schemas/__test__/fixtures';

export const versionPayload = [
  schemaVersion1,
  schemaVersion2,
  schemaVersionWithNonAsciiChars,
];
export const versionEmptyPayload = [];

export const jsonSchema: SchemaSubject = {
  subject: 'test',
  version: '15',
  id: 1,
  schema:
    '{"type":"record","name":"MyRecord1","namespace":"com.mycompany","fields":[{"name":"id","type":"long"}]}',
  compatibilityLevel: 'BACKWARD',
  schemaType: SchemaType.JSON,
};

export const protoSchema: SchemaSubject = {
  subject: 'test_proto',
  version: '1',
  id: 2,
  schema:
    'syntax = "proto3";\npackage com.indeed;\n\nmessage MyRecord {\n  int32 id = 1;\n  string name = 2;\n}\n',
  compatibilityLevel: 'BACKWARD',
  schemaType: SchemaType.PROTOBUF,
};
