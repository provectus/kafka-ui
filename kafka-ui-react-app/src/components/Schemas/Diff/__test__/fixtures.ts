import { SchemaSubject, SchemaType } from 'generated-sources';

export const versions: SchemaSubject[] = [
  {
    subject: 'test',
    version: '3',
    id: 3,
    schema:
      'syntax = "proto3";\npackage com.indeed;\n\nmessage MyRecord {\n  int32 id = 1;\n  string name = 2;\n}\n',
    compatibilityLevel: 'BACKWARD',
    schemaType: SchemaType.PROTOBUF,
  },
  {
    subject: 'test',
    version: '2',
    id: 2,
    schema:
      '{"type":"record","name":"MyRecord2","namespace":"com.mycompany","fields":[{"name":"id","type":"long"}]}',
    compatibilityLevel: 'BACKWARD',
    schemaType: SchemaType.JSON,
  },
  {
    subject: 'test',
    version: '1',
    id: 1,
    schema:
      '{"type":"record","name":"MyRecord1","namespace":"com.mycompany","fields":[{"name":"id","type":"long"}]}',
    compatibilityLevel: 'BACKWARD',
    schemaType: SchemaType.JSON,
  },
];
