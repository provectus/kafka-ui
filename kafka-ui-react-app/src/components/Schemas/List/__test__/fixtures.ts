import { SchemaSubject } from 'generated-sources';

export const schemas: SchemaSubject[] = [
  {
    subject: 'test',
    version: '1',
    id: 1,
    schema:
      '{"type":"record","name":"MyRecord1","namespace":"com.mycompany","fields":[{"name":"id","type":"long"}]}',
    compatibilityLevel: 'BACKWARD',
  },
  {
    subject: 'test2',
    version: '1',
    id: 1,
    schema:
      '{"type":"record","name":"MyRecord2","namespace":"com.mycompany","fields":[{"name":"id","type":"long"}]}',
    compatibilityLevel: 'BACKWARD',
  },
  {
    subject: 'test3',
    version: '1',
    id: 1,
    schema:
      '{"type":"record","name":"MyRecord3","namespace":"com.mycompany","fields":[{"name":"id","type":"long"}]}',
    compatibilityLevel: 'BACKWARD',
  },
];
