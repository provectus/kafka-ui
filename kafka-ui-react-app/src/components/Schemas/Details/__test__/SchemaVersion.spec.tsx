import React from 'react';
import { shallow } from 'enzyme';
import SchemaVersion from '../SchemaVersion';
import { versions } from './fixtures';

describe('SchemaVersion', () => {
  it('renders versions', () => {
    const wrapper = shallow(<SchemaVersion version={versions[0]} />);

    expect(wrapper.find('td').length).toEqual(3);
    expect(wrapper.exists('JSONViewer')).toBeTruthy();
  });

  it('matches snapshot', () => {
    expect(shallow(<SchemaVersion version={versions[0]} />)).toMatchSnapshot();
  });
});
