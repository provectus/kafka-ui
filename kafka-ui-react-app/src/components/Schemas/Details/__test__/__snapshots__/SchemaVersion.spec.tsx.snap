// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SchemaVersion matches snapshot 1`] = `
<tr>
  <td>
    1
  </td>
  <td>
    1
  </td>
  <td
    className="py-0"
  >
    <JSONViewer
      data={
        Object {
          "fields": Array [
            Object {
              "name": "id",
              "type": "long",
            },
          ],
          "name": "MyRecord1",
          "namespace": "com.mycompany",
          "type": "record",
        }
      }
    />
  </td>
</tr>
`;
